support-frontend/assets/components/csr/csrMode.js

```diff
@@ -0,0 +1,17 @@
+
+const domains = ["https://gnmtouchpoint.my.salesforce.com", "https://gnmtouchpoint--dev1--c.cs88.visual.force.com"]
+const isSalesforceDomain = domain => domains.find(element => element === domain)
+
+const isInCsrMode = () => window.location !== window.parent.location;
+
+const listenForCsrDetails = callback =>
+  window.addEventListener(
+    'message',
+    (event) => {
+      if (isSalesforceDomain(event.origin)) {
+        callback(event.data);
+      }
+    },
+  );
```

Comment:
I think this could be simplified if implemented as a custom hook, which also gives us easy cleanup of the event listener on `window`- something like
```suggestion
import { useEffect, useState } from 'react';

export function useCsrDetails() {
	const [csrUsername, setCsrUsername] = useState('');
	
	function checkForParentMessage (event) {
		if (isSalesforceDomain(event.origin)) {
	    	setCsrUsername(event.data);
	    }
	}

    useEffect(() => {
	    window.addEventListener('message', checkForParentMessage);
	    return () => window.removeEventListener('message', checkForParentMessage);
    }, []);

    return csrUsername;
}
```

and then used like

```javascript
const CsrBanner = () => {
  const csrUsername = useCsrDetails();

//etc
```

Reply:
I like this suggestion @i-hardy. Could we then do away with the  `isInCsrMode` export and instead have a conditional in `CsrBanner` that checks `csrUsername`, which the default value is `''` and therefore falsy eg.

```js
const CsrBanner = () => {
  const csrUsername = useCsrDetails();

  return csrUsername ? (
      <div css={container}>
        <div css={banner}>You are in customer support mode. Signed in as: {csrUsername}</div>
      </div>
    ) : null;
};
```

Reply:
I've updated to Imogen's suggestion. The reason that these are separated out into different function in a different module is that I know they are going to also be needed in the checkout to get details of the currently signed in CSR. I haven't done that work in this PR though as this is a spike for now.